src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.cs

```diff
@@ -68,16 +173,58 @@ string IDkmClrFormatter.GetUnderlyingString(DkmClrValue value, DkmInspectionCont
         internal abstract bool IsWhitespace(char c);
 
         // Note: We could be less conservative (e.g. "new C()").
-        internal bool NeedsParentheses(string expr)
+        private bool NeedsParentheses(string expr)
         {
-            foreach (var ch in expr)
+            int parens = 0;
+            for (int i = 0; i < expr.Length; i++)
             {
-                if (!this.IsIdentifierPartCharacter(ch) && !this.IsWhitespace(ch) && ch != '.') return true;
+                var ch = expr[i];
+                switch (ch)
+                {
+                    case '(':
+                        // Cast, "(A)b", requires parentheses.
+                        if ((parens == 0) && FollowsCloseParen(expr, i))
```

Comment:
The parens may be from a call. The string can be an arbitrary C# expression, entered explicitly by the user or generated as a fullname by an EE implementation.